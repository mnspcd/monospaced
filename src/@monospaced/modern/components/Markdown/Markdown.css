/*
 * Run-in

.Markdown > p:first-of-type::first-line {
  font-family: var(--font-small-caps);
}
 */

/*
 * Spacing
 */

.mdrn .Markdown blockquote,
.mdrn .Markdown ol,
.mdrn .Markdown p,
.mdrn .Markdown ul {
  margin-bottom: calc(1em * var(--leading) / 2);
}

.mdrn .Markdown h1 {
  margin-bottom: calc(1em * var(--leading) / var(--type-scale-xxxl));
}

.mdrn .Markdown h2,
.mdrn .Markdown h3,
.mdrn .Markdown h4,
.mdrn .Markdown h5,
.mdrn .Markdown h6 {
  margin-bottom: calc(12rem / 16);
  margin-top: calc(24rem / 16);
}

@media (--larger-viewport) {
  .mdrn .Markdown h2,
  .mdrn .Markdown h3,
  .mdrn .Markdown h4,
  .mdrn .Markdown h5,
  .mdrn .Markdown h6 {
    margin-bottom: calc(15rem / 16);
    margin-top: calc(30rem / 16);
  }
}

.mdrn .Markdown hr,
.mdrn .Markdown pre,
.mdrn .Markdown table {
  margin-bottom: calc(1em * var(--leading));
  margin-top: calc(1em * var(--leading));
}

.mdrn .Markdown > :first-child {
  margin-top: 0;
}

.mdrn .Markdown > :last-child {
  margin-bottom: 0;
}

/*
 * Small block
 */

.mdrn .Markdown > small {
  display: block;
  margin-bottom: calc(1em * var(--leading) / 2);
  max-width: var(--measure);
}

@media (--larger-viewport) {
  .mdrn .Markdown > small {
    margin-bottom: calc(1em * var(--leading) / var(--type-scale-s) / 2);
    max-width: calc(var(--measure) * var(--type-scale-l));
  }
}

/*
 * Footnotes
 */

@media (--larger-viewport) {
  .mdrn .Markdown .footnotes ol {
    font-size: calc(1em * var(--type-scale-s));
  }
}

.mdrn .Markdown .footnotes p {
  margin-bottom: 0;
}

.mdrn .Markdown .footnote-backref {
  display: block;
  font-family: var(--font-monospaced);
  font-size: calc(1em * var(--type-scale-xl));
  line-height: 1;
}
