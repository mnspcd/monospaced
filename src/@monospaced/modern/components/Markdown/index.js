import React from "react";

import Blockquote from "../Blockquote";
import Code from "../Code";
import Heading from "../Heading";
import List from "../List";
import Paragraph from "../Paragraph";
import Rule from "../Rule";
import Table from "../Table";

const Pre = props => <pre {...props} />;

const mdxComponents = {
  blockquote: Blockquote,
  code: Code.Block,
  h1: Heading.H1,
  h2: Heading.H2,
  h3: Heading.H3,
  h4: Heading.H4,
  h5: Heading.H5,
  h6: Heading.H6,
  hr: Rule,
  inlineCode: Code,
  ol: List.Ordered,
  p: Paragraph,
  pre: Pre,
  table: Table,
  ul: List,
};

export { default } from "./Markdown";
export { mdxComponents };
