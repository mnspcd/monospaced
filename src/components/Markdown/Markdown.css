/*
 * Run-in

.Markdown > div > p:first-of-type::first-line {
  font-family: var(--font-small-caps);
}
 */

/*
 * Spacing
 */

.Markdown blockquote,
.Markdown ol,
.Markdown p,
.Markdown ul {
  margin-bottom: calc(1em * var(--leading) / 2);
}

.Markdown h1 {
  margin-bottom: calc(24rem / 16);
}

@media (--larger-viewport) {
  .Markdown h1 {
    margin-bottom: calc(30rem / 16);
  }
}

.Markdown h2,
.Markdown h3,
.Markdown h4,
.Markdown h5,
.Markdown h6 {
  margin-bottom: calc(12rem / 16);
  margin-top: calc(24rem / 16);
}

@media (--larger-viewport) {
  .Markdown h2,
  .Markdown h3,
  .Markdown h4,
  .Markdown h5,
  .Markdown h6 {
    margin-bottom: calc(15rem / 16);
    margin-top: calc(30rem / 16);
  }
}

.Markdown hr,
.Markdown pre,
.Markdown table {
  margin-bottom: calc(1em * var(--leading));
  margin-top: calc(1em * var(--leading));
}

.Markdown > div > :first-child {
  margin-top: 0;
}

.Markdown > div > :last-child {
  margin-bottom: 0;
}

/*
 * Footnotes
 */

@media (--larger-viewport) {
  .Markdown .footnotes ol {
    font-size: calc(1em * var(--type-scale-s));
  }
}

.Markdown .footnotes p {
  margin-bottom: 0;
}

.Markdown .footnote-backref {
  display: block;
  font-family: var(--font-monospaced);
  font-size: calc(1em * var(--type-scale-xl));
  line-height: 1;
}
