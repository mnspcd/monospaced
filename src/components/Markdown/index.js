import Blockquote from "./components/Blockquote";
import Code from "./components/Code";
import Heading from "./components/Heading";
import List from "./components/List";
import Paragraph from "./components/Paragraph";
import Rule from "./components/Rule";
import Table from "./components/Table";

const components = {
  blockquote: Blockquote,
  code: Code.Block,
  h1: Heading.H1,
  h2: Heading.H2,
  h3: Heading.H3,
  h4: Heading.H4,
  h5: Heading.H5,
  h6: Heading.H6,
  hr: Rule,
  inlineCode: Code,
  ol: List.Ordered,
  p: Paragraph,
  table: Table,
  ul: List,
};

export { default } from "./Markdown";
export { components, Blockquote, Code, Heading, List, Paragraph, Rule, Table };
