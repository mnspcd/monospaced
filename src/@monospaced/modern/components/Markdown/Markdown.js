import PropTypes from "prop-types";
import React from "react";

import "./Markdown.css";

/**
 * Use `Markdown` as a layout wrapper for Mdx documents.
 */
const Markdown = ({ children }) => <div className="Markdown">{children}</div>;

Markdown.propTypes = { children: PropTypes.node.isRequired };

export default Markdown;
