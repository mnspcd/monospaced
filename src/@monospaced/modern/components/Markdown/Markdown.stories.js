import { MDXProvider } from "@mdx-js/react";
import { select } from "@storybook/addon-knobs";
import React from "react";

import Grid from "../Grid";
import Space from "../Space";
import Surface from "../Surface";

import Markdown, { mdxComponents } from "./";
import Mdx from "./__mocks__/basic.mock.mdx";
import MdxSample from "./__mocks__/sample.mock.mdx";

export default {
  component: Markdown,
  title: "Components/Markdown",
};

export const basic = () => (
  <MDXProvider components={mdxComponents}>
    <Markdown>
      <Mdx />
    </Markdown>
  </MDXProvider>
);

export const sample = () => (
  <MDXProvider components={mdxComponents}>
    <Markdown>
      <MdxSample />
    </Markdown>
  </MDXProvider>
);

export const knobs = () => (
  <Surface
    backgroundColor={select(
      "backgroundColor",
      Surface.backgroundColors,
      "white",
      "Surface",
    )}
  >
    <Grid>
      <Grid.Item colSpan="4">
        <Space />
        <MDXProvider components={mdxComponents}>
          <Markdown>
            <MdxSample />
          </Markdown>
        </MDXProvider>
      </Grid.Item>
    </Grid>
  </Surface>
);

knobs.story = { parameters: { docs: { disable: true } } };
