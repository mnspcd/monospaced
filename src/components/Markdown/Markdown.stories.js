import React from "react";
import { select, text } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";

import Mdx from "./__mocks__/markdown.mock.mdx";
import Surface from "../Surface";
import { components } from "../Markdown";

storiesOf("Components/Markdown", module).add("Sample", () => (
  <Surface
    backgroundColor={select(
      "backgroundColor",
      Surface.backgroundColors,
      "white",
      "Surface",
    )}
    padding={text("padding", "", "Surface")}
  >
    <Mdx components={components} />
  </Surface>
));
