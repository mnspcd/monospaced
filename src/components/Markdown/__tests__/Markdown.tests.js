/* eslint-env jest */
import React from "react";
import { render } from "enzyme";

import Markdown from "../Markdown";

describe("Markdown component", () => {
  it("should render correctly", () => {
    const component = render(<Markdown>children</Markdown>);
    expect(component).toMatchSnapshot();
  });
});
