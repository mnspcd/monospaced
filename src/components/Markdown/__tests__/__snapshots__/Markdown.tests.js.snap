// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown component should render correctly 1`] = `
<div
  class="Markdown"
>
  children
</div>
`;
